x-pack/plugins/event_log/server/es/context.test.ts

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { createEsContext } from './context';
+import { ClusterClient, Logger } from '../../../../../src/core/server';
+import { elasticsearchServiceMock, loggingServiceMock } from '../../../../../src/core/server/mocks';
+
+type EsClusterClient = Pick<jest.Mocked<ClusterClient>, 'callAsInternalUser' | 'asScoped'>;
+
+let logger: Logger;
+let clusterClient: EsClusterClient;
+
+beforeEach(() => {
+  logger = loggingServiceMock.createLogger();
+  clusterClient = elasticsearchServiceMock.createClusterClient();
+});
+
+describe('createEsContext', () => {
+  test('should return is ready state as falsy if not initialized', () => {
+    const context = createEsContext({
+      logger,
+      clusterClientPromise: Promise.resolve(clusterClient),
+      indexNameRoot: 'test0',
+    });
+
+    expect(context.initialized).toBeFalsy();
+
+    context.initialize();
+    expect(context.initialized).toBeTruthy();
+  });
+
+  test('should return esNames', () => {
+    const context = createEsContext({
+      logger,
+      clusterClientPromise: Promise.resolve(clusterClient),
+      indexNameRoot: 'test-index',
+    });
+
+    const esNames = context.esNames;
+    expect(esNames).toStrictEqual({
```

Comment:
We may have to mock package.json like [here](https://github.com/elastic/kibana/blob/master/x-pack/plugins/event_log/server/es/names.test.ts#L9-L11) to avoid breaking tests whenever backporting or bumping versions.